core/src/main/java/org/wso2/msf4j/internal/router/MicroserviceMetadata.java

```diff
@@ -73,10 +73,7 @@ public MicroserviceMetadata(Map<String, Object> services) {
                     patternRouter.add(absolutePath, new HttpResourceModel(absolutePath, method, service, false));
                 } else if (Modifier.isPublic(method.getModifiers()) && method.getAnnotation(Path.class) != null) {
                     // Sub resource locator method
-                    String relativePath = method.getAnnotation(Path.class).value();
-                    if (relativePath.startsWith("/")) {
-                        relativePath = relativePath.substring(1);
-                    }
+                    String relativePath = Utils.normalizePath(method.getAnnotation(Path.class).value());
```

Comment:
Shall we normalize in the patternRouter.add method only.

We already do a normalisation in the patternRouter.add method (1st 2 lines of the method are as shown below)

```java
        // replace multiple slashes with a single slash.
        String path = source.replaceAll("/+", PATH_SLASH);

        path = (path.endsWith(PATH_SLASH) && path.length() > 1)
                ? path.substring(0, path.length() - 1) : path;
```
Instead of this we can use the new normalize method.

Then we do not need to normalise here and it will gain us a little bit of performance :)

Reply:
Hmm, then why did the original code tried to strip a leading "/'?  Also looks to me the relativePath comes directly from the annotation (e.g. `@Path("/sub")`), so I don't understand your "is already normalized".  Seems I have to read the code further...

Reply:
I think you misunderstood my comment. patternRouter.add method will normalise the path before proceeding with its logic (with this PR using the new normalising method). The original code which was here for stripping the leading '/' actually is not necessary previously as well and should have been removed earlier.
Since we are anyway normalising the path in patternRouter.add method, we do not need to normalise the path in line 76 at all.

WDYT?